src/include/geometry/Point.h

```diff
@@ -11,140 +11,180 @@ namespace cuhksz {
+		Point& ret = *(new Point);
+		for (int i = 0; i < N; i ++)
+			ret.elem[i] = elem[i] - other.elem[i];
+		return ret;
+	}
+
+	Point& operator+=(Point const &other) {
+		for (int i = 0; i < N; i ++)
+			elem[i] += other.elem[i];
+        return *this;
+	}
+
+	Point& operator-=(Point const &other) {
+		for (int i = 0; i < N; i ++)
+			elem[i] -= other.elem[i];
+        return *this;
+	}
+
+	// dot product
+	double operator*(Point const &other) const {
+		double ret = 0.0;
+		for (int i = 0; i < N; i ++)
+			ret += elem[i] * other.elem[i];
+		return ret;
+	}
+
+	Point& operator*(double const &constant) const {
+		Point& ret = *(new Point);
+		for (int i = 0; i < N; i ++)
+			ret.elem[i] = elem[i] * constant;
+		return ret;
+	}
+
+    friend Point operator*(double const &constant, Point const &self) {
+        return self * constant;
+    }
+
+	Point& operator*=(double const &constant) {
+		for (int i = 0; i < N; i ++)
+			elem[i] *= constant;
+        return *this;
+	}
+
+	Point& operator/(double const &constant) const {
+		Point& ret = *(new Point);
+		for (int i = 0; i < N; i ++)
+			ret.elem[i] = elem[i] / constant;
+		return ret;
+	}
+
+	Point& operator/=(double const &constant) {
+		for (int i = 0; i < N; i ++)
+			elem[i] /= constant;
+        return *this;
+	}
+
+	bool operator==(Point const &other) const {
+		for (int i = 0; i < N; i ++)
+			if (fabs(elem[i] - other.elem[i]) < EPS)
+				return false;
+		return true;
+	}
+
+	bool operator!=(Point const &other) const {
+		return !(*this == other);
+	}
+
+	bool operator<(Point const &other) const {
+		for (int i = 0; i < N; i ++)
+			if (elem[i] != other.elem[i])
+				return elem[i] < other.elem[i];
+		return false;
+	}
+
+	bool operator<=(Point const &other) const {
+		for (int i = 0; i < N; i ++)
+			if (elem[i] != other.elem[i])
+				return elem[i] < other.elem[i];
+		return true;
+	}
+
+	bool operator>(Point const &other) const {
+		for (int i = 0; i < N; i ++)
+			if (elem[i] != other.elem[i])
+				return elem[i] > other.elem[i];
+		return false;
+	}
+
+	bool operator>=(Point const &other) const {
+		for (int i = 0; i < N; i ++)
+			if (elem[i] != other.elem[i])
+				return elem[i] > other.elem[i];
+		return true;
+	}
+
+	friend std::ostream& operator<<(std::ostream& os, Point const self) {
+		os << "(" << self.elem[0];
+		for (int i = 1; i < N; i ++)
+			os << ", " << self.elem[i];
+		return os << ")";
+	}
+
+    // TODO: cross product
+	friend double cross(const Point& v, const Point& w) {
+		static_assert(N == 2, "cross product only apply to 2D");
+		return v.elem[0] * w.elem[1] - v.elem[1] * w.elem[0];
+	}
+
+	double len() {
+		double ret = 0.0;
+		for (int i = 0; i < N; i ++)
+			ret += elem[i] * elem[i];
+		return sqrt(ret);
+	}
+
+    void normalize() {
+        double l = len();
+        for (int i = 0; i < N; i ++)
+            elem[i] /= l;
+    }
+
+private:
+	double elem[N];	// coordinates
 };
 
-}  // namespace cuhksz
+#define Vector Point
```

Comment:
Consider the name conflict if user `using namespace cuhksz` and both include `Point.h` and `Vector.h`. Also better to use `using` or `typedef` for type alias. Consider name `GVector` to avoid the conflicts